### PR TITLE
Separate the normal pycodestyle entry point from the ci run parametres

### PR DESCRIPTION
**Pull request by Rotonen**
This allows us to ship a shared config file and also a normal entry point for editor integrations.

I'm not 100% if I lost corner cases with the translation to just a flat shell script.

### CODE REVIEW

File: test-base.cfg [hunk 1]
```diff
@@ -189,23 +190,28 @@ output = ${buildout:directory}/bin/package-directory
...
+eggs = pycodestyle
 
 # Downloads the default pycodestyle.cfg.
 [pycodestyle-cfg]
 recipe = collective.recipe.shelloutput
 cfgfile = ${buildout:directory}/parts/pycodestyle/pycodestyle.cfg
 cfgurl = https://raw.githubusercontent.com/4teamwork/ftw-buildouts/master/pycodestyle.cfg
 commands =
-    cmd1 = mkdir -p `dirname ${pycodestyle-cfg:cfgfile}`
+    cmd1 = mkdir -p $(dirname ${pycodestyle-cfg:cfgfile})
```
Comment (tarnap) [inline test-base.cfg]: <3

File: test-base.cfg [hunk 2]
```diff
@@ -215,7 +221,7 @@ initialization =
         shell=True, stdout=subprocess.PIPE).stdout.read().strip()
     if '--absolute' in sys.argv: pkgdir = os.path.abspath(pkgdir) ; sys.argv.remove('--absolute')
     if len(sys.argv) < 2: sys.argv.extend([pkgdir])
-    sys.argv.extend(${pycodestyle:args})
+    sys.argv.extend(${pycodestyle-ci-run:args})
```
Comment (jone) [inline test-base.cfg]: There's something odd now: the `initialization` option is now in the `[pycodestyle-ci-run]` section, but this won't be used by the `collective.recipe.template`.

I think `bin/pycodestyle` does no longer include the intialization code and therefore does no longer do the same thing as before.

`bin/pycodestyle` should work the same as before (it should limit to the source directory of this package).

Comment (Rotonen) [inline test-base.cfg]: The whole point is that `bin/pycodestyle` should offer the normal entry point of the `pycodestyle` egg so editors and other tools can interface with in the ways they expect ('run the tool on this buffer from some tmp while the user is still typing').

Once that's the normal entry point, the CI-run one should just be a simple shell script with a few extra parametres.

Comment (jone) [inline test-base.cfg]: In my emacs configuration I run `./bin/pycodestyle --absolute`, expecting the source of the current package to be scanned. Do I need to switch to `bin/pycodestyle-ci-run` then?
I really don't want all dependencies in `./src` do be scanned, nor do I want `parts/omlette` to be scanned.

And:
```
$ ./bin/buildout
warning: no files found matching '*.txt'
warning: no previously-included files matching '._*' found anywhere in distribution
warning: no previously-included files matching '*.mo' found anywhere in distribution
While:
  Installing.
  Getting section pycodestyle-ci-run.
  Initializing section pycodestyle-ci-run.
  Getting option pycodestyle-ci-run:initialization.
Error: Referenced option does not exist: pycodestyle-ci-run args
```

Comment (Rotonen) [inline test-base.cfg]: In your particular built up history of how you've done things, yes. Or you have a post-save hook for running `bin/pycodestyle` on the file which just got saved. (Or some advanced magic with the check-while-you-type buffer flushes.)

Comment (jone) [inline test-base.cfg]: I actually realized that I also used to do `bin/pep8` on the terminal quite often, order to see all violations within the current package. Would be great to not loose that.

Comment (Rotonen) [inline test-base.cfg]: The best I would offer for your workflow is reverse-i-search `^R pycode`.

The core issue at hand here is that editors and IDEs blindly assume the executable name.

Comment (Rotonen) [inline test-base.cfg]: I missed I had accidentally left in the old init section into the ci run script. My bad.

**Comment by jone** [comment]
I have realized that my emacs integration does no longer work with this change, no matter what script I use.
If I setup my emacs with `bin/pycodestyle` I don't have our pycodestyle-config-file anymore, if I use `pycodestyle-ci-run` I'm not able to check a single file as arguments are not passed along.

I have these requirements:
1. run the checker preconfigured against one file with `bin/xy path/to.py`
2. run the checker prefconfigured against all python files which are considered part of the repository with `bin/xy`
3. 1 and 2 should already come with the config file preconfigured
4. 1 and 2 may implemented in separate scripts, names don't matter

How can we keep backwards compatibility while moving forward?

**Comment by Rotonen** [comment]
That means we need to bundle the `setup.cfg` generation into this buildout as well. That config will be picked up by pycodestyle.

Ideas on how to make a bit of buildout to concatenate files, just use a pile of pure shell with soft fails for config files not there?

**Comment by jone** [comment]
When we update the `setup.cfg` in buildout we must think about how to handle the case where there is already a `setup.cfg` in the repository, [although this is rare](https://github.com/search?utf8=%E2%9C%93&q=org%3A4teamwork+filename%3Asetup.cfg&type=Code).

In order to move forward here, I suggest that we add a `bin/pycodestyle-package` with the old behavior (for backwards compatibility), so that we can merge your new changes and move forward.
Something like:
```ini
[pycodestyle-package]
recipe = collective.recipe.template
output = ${buildout:bin-directory}/pycodestyle-package
mode = 755
input = inline:
    #!/bin/bash
    if [ $# -eq 0 ]; then
      paths=$(${buildout:directory}/bin/package-directory relative)
    else
      paths=$@
    fi
    ${buildout:directory}/bin/pycodestyle --exclude=tests --config="${pycodestyle-cfg:cfgfile}" $paths
```

**Comment by Rotonen** [comment]
@jone the wrapper script entry point can also be interfaced with, this pull request has now become unnecessary.
